app/lib/actions/katello/event_queue/monitor.rb

```diff
@@ -0,0 +1,60 @@
+module Actions
+  module Katello
+    module EventQueue
+      class Monitor < Actions::EntryAction
```

Comment:
Iirc foreman-maintain performs checks for long running tasks before upgrade, we'll probably need to whitelist this new action in there.